RNA-seq pipeline
================

Sequence reads were aligned to a transcriptome reference generated by JAGuar using mm10v38 reference genome annotated with read-length-specific exon-exon junction sequences. To quantify gene expression, reads per kilobase per million mapped reads (RPKM) were calculated. Both protein-coding and non-coding RNA were included within the analyses. Differentially expressed genes were called using a custom in-house matlab tool (DEFine, FDR cutoff = 0.05; RPKM > 0.01), which performs pair-wise comparisons by normalizing expression using GC content, followed by subsequent binning of similarly expressed genes together to determine the genes that varied the greatest within each bin.

1) [join_nc_pc_genes.sh](https://github.com/Jwong684/bioinformatics/tree/master/analyses/RNAseq/join_nc_pc_genes.sh)

- combines non-coding and protein-coding genes into one file.

2) [DEfine.m](https://github.com/Jwong684/bioinformatics/tree/master/analyses/RNAseq/DEfine.m)

- uses an in-house script (called DEfine designed by our bioinformatician Misha Bilenky) to compare any two RNA-seq libraries (using the product of join_nc_pc_genes.sh)
- parameters to modify are names and directory.
- make a separate script for every pairwise comparison performed.

3) [runall.sh](https://github.com/Jwong684/bioinformatics/tree/master/analyses/RNAseq/runall.sh)

- runs all DEfine.m scripts in a matlab screen.

4) [extractGene.sh](https://github.com/Jwong684/bioinformatics/tree/master/analyses/RNAseq/extractGene.sh)

- takes the list of DEG output of DEfine, and matches Ensembl ID with Gene ID.

Output: ens \t gene \t rpkm_1 \t rpkm_2 \t q-value (FDR)

5) [DEG_summary.sh](https://github.com/Jwong684/bioinformatics/tree/master/analyses/RNAseq/DEG_summary.sh)

- compiles all of the DEGs and summarizes it into a table

Output: number of DEG \t UP or DN \t timepoint \t cell-type